provider/datagen/src/transform/cldr/units/info.rs

```diff
@@ -0,0 +1,136 @@
+            });
+
+            let units_info_index = UnitsInfoIndex {
+                quantity: Some(quantity_index as u16),
+                convert_unit: None,
+            };
+
+            conversion_info_map.insert(unit_name.as_str(), units_info_index);
+        }
+
+        let mut convert_units_vec = Vec::<ConvertUnits>::new();
+        for (unit_name, convert_unit) in convert_units {
+            let base_unit = convert_unit.base_unit.as_str();
+            let factor = match convert_unit.factor {
+                Some(ref factor) => factor.as_str(),
+                None => "1",
+            };
+
+            let convert_unit_index = convert_units_vec.len();
+            convert_units_vec.push(ConvertUnits {
+                base_unit: Cow::Borrowed(base_unit),
+                factor: Cow::Borrowed(factor),
+            });
+
+            if let Some(units_info_index) = conversion_info_map.get_mut(unit_name.as_str()) {
+                units_info_index.convert_unit = Some(convert_unit_index as u16);
+            } else {
+                let units_info_index = UnitsInfoIndex {
+                    quantity: None,
+                    convert_unit: Some(convert_unit_index as u16),
+                };
+                conversion_info_map.insert(unit_name.as_str(), units_info_index);
+            }
+        }
+
+        let result = UnitsInfoV1 {
+            units_info: ZeroMap::from_iter(conversion_info_map),
+            unit_quantity: VarZeroVec::from(&quantity_vec),
+            convert_units: VarZeroVec::from(&convert_units_vec),
+        };
+
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(result)),
+        })
+    }
+}
+
+impl IterableDataProvider<UnitsInfoV1Marker> for crate::DatagenProvider {
+    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+        Ok(vec![Default::default()])
+    }
+}
+
+#[test]
+fn test_basic() {
+    use icu_locid::locale;
+    use icu_provider::prelude::*;
+    use zerovec::maps::ZeroVecLike;
+
+    let provider = crate::DatagenProvider::new_testing();
+
+    let und: DataPayload<UnitsInfoV1Marker> = provider
+        .load(DataRequest {
+            locale: &locale!("und").into(),
+            metadata: Default::default(),
+        })
+        .unwrap()
+        .take_payload()
+        .unwrap();
+
+    let units_info = und.get().to_owned();
+    let units_info_map = &units_info.units_info;
+    let unit_quantity = &units_info.unit_quantity;
+    let convert_units = &units_info.convert_units;
+
+    let meter = units_info_map.get("meter").unwrap();
+    let meter_quantity_index = meter.quantity.get().unwrap().as_unsigned_int() as usize;
+    let meter_quantity = unit_quantity.zvl_get(meter_quantity_index).unwrap();
+    assert_eq!(&meter_quantity.quantity, "length");
+    // TODO: how to test this?
+    // assert_eq!(meter_quantity.constant_exactness as u8, QuantitySimplicity::Simple as u8);
+
```

Comment:
@Manishearth : how to test `QuantitySimplicityULE` and `QuantitySimplicity`. Both of them are enums.

Reply:
Call `QuantitySimplicity::from_unaligned(ule)` on the ULE value to turn it into the real enum.